core/templates/pages/exploration-editor-page/services/exploration-rights.service.spec.ts

```diff
@@ -280,6 +280,44 @@ describe('Exploration rights service', () => {
     expect(ers.viewerNames).toEqual(serviceData.rights.viewer_names);
   }));
 
+  it('should remove user from UI before updating with backend data', fakeAsync(() => {
+    ers.editorNames = ['editorName', 'newEditor'];
+    serviceData.rights.editor_names = [];
+    spyOn(
+      explorationRightsBackendApiService,
+      'removeRoleAsyncDeleteData'
+    ).and.returnValue(Promise.resolve(serviceData));
+
+    ers.removeRoleAsync('editorName').then(successHandler, failHandler);
+    expect(ers.editorNames).toEqual(['newEditor']);
+    expect(ers.editorNames).not.toEqual(serviceData.rights.editor_names);
+    tick();
+
+    expect(successHandler).toHaveBeenCalled();
+    expect(failHandler).not.toHaveBeenCalled();
+
+    expect(ers.editorNames).toEqual(serviceData.rights.editor_names);
+  }));
+
+  it('should redisplay users when removing a user fails', fakeAsync(() => {
+    ers.ownerNames = ['ownerName'];
+    ers.viewerNames = ['viewerName'];
+    spyOn(
+      explorationRightsBackendApiService,
+      'removeRoleAsyncDeleteData'
+    ).and.returnValue(Promise.reject(new Error('Error')));
+
+    ers.removeRoleAsync('ownerName').then(successHandler, failHandler);
+    expect(ers.ownerNames).not.toEqual(['ownerName']);
```

Comment:
In general, try to assert equality rather than inequality (it's a stronger check).

Reply:
Done